diff --git a/pyproject.toml b/pyproject.toml
index original..modified 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,8 @@ include = ["nhentai/viewer/**"]
 
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.9"
+curl-cffi = "^0.13.0"
 requests = "^2.32.3"
 soupsieve = "^2.6"
 beautifulsoup4 = "^4.12.3"
diff --git a/nhentai/utils.py b/nhentai/utils.py
index a627537..44ad764 100644
--- a/nhentai/utils.py
+++ b/nhentai/utils.py
@@ -11,6 +11,7 @@ import requests
 import sqlite3
 import urllib.parse
 from typing import Tuple
+from curl_cffi import requests as cffi_requests
 
 from nhentai import constant
 from nhentai.constant import PATH_SEPARATOR
@@ -36,7 +37,7 @@ def get_headers():
     return headers
 
 def request(method, url, **kwargs):
-    session = requests.Session()
+    session = cffi_requests.Session(impersonate="chrome110")
     session.headers.update(get_headers())
 
     if not kwargs.get('proxies', None):
